Ws-Peroth/20211007/MapGeneratorBSP.cs

```diff
@@ -0,0 +1,251 @@
+        // depth  : 깊이
+        var lengthX = endX - startX;
+        var lengthY = endY - startY;
+
+        // 깊이가 0 이거나 사용 가능한 공간이 10 이하일 경우에 방을 생성
+        if (depth == 0 || lengthX <= 10 || lengthY <= 10)
+        {
+            return GenerateRoom(startX, startY, endX, endY);
+        }
+        MapLocationPosition firstRoom, secondRoom;
+        int dividenum;
+
+        // X의 길이가 더 길 경우 세로방향으로 분할
+        if (lengthX > lengthY)
+        {
+            dividenum = RoomDivideRatio * lengthX / 10;
+
+            // x 좌표의 끝을 현재 x + 나눌 범위 로 갱신 
+            firstRoom = DivideMap(depth - 1, startX, startY, startX + dividenum, endY);
+
+            // x 좌표의 시작부분을 현재 x + 나눌 범위로 갱신
+            secondRoom = DivideMap(depth - 1, startX + dividenum, startY, endX, endY);
+
+            // 분할한 두 방을 병합
+            MergeVerticalRoom(firstRoom, secondRoom);
+        }
+        else    // 가로 방향으로 분할
+        {
+            dividenum = RoomDivideRatio * lengthY / 10;
+
+            // y 좌표의 끝을 현재 y + 나눌 범위로 갱신
+            firstRoom = DivideMap(depth - 1, startX, startY, endX, startY + dividenum);
+
+            // y 좌표의 시작 부분을 현재 y + 나눌 범위로 갱신
+            secondRoom = DivideMap(depth - 1, startX, startY + dividenum, endX, endY);
+
+            // 분할한 두 방을 병합
+            MergeHorizonRoom(firstRoom, secondRoom);
+        }
+
+        return new MapLocationPosition(firstRoom.startX, firstRoom.startY, firstRoom.endX, firstRoom.endY);
+    }
+
+    private MapLocationPosition GenerateRoom(int startX, int startY, int endX, int endY)
+    {
+        // (startX, startY) ~ (endX, endY) 의 구역 중, border만큼을 띄우고 방을 생성
+        for (var y = startY + _borderSizeY; y < endY - _borderSizeY; y++)
+        {
+            for (var x = startX + _borderSizeX; x < endX - _borderSizeX; x++)
+            {
+                map[y, x] = 0;
+            }
+        }
+
+        // 실제 방의 위치를 반환함
+        // 방이 생성된 위치는 border의 안쪽임으로, border의 크기만큼 연산을 하여 반환
+        return new MapLocationPosition
+            (
+                startX + _borderSizeX,
+                startY + _borderSizeY,
+                endX - _borderSizeX - 1,
+                endY - _borderSizeY - 1
+            );
+    }
+
+    private void MergeVerticalRoom(MapLocationPosition firstRoom, MapLocationPosition secondRoom)
+    {
+        // 생성할 통로의 넓이
+        int wide;
+
+        // 임의의 넓이를 받아옴
+        wide = GetRandomVerticalWide(0, _wideMax, firstRoom.startY, firstRoom.endY);
+
+        // 두 구역의 시작부분끼리 이어진 통로를 생성
+        ConnectVerticalRoom(firstRoom.endX, secondRoom.startX, firstRoom.startY, wide);
+
+        // 임의의 넓이를 받아옴
+        wide = GetRandomVerticalWide(0, _wideMax, firstRoom.startY, firstRoom.endY);
+
+        // 두 구역의 끝부분끼리 이어진 통로를 생성
+        ConnectVerticalRoom(firstRoom.endX, secondRoom.startX, firstRoom.endY, wide);
+    }
+
+    private int GetRandomVerticalWide(int min, int max, int startLimit, int endLimit, int defaultValue = 0)
+    {
+        // 기본값 설정
+        var wide = defaultValue;
```

Comment:
```suggestion
        int wide;
```
~~어차피 for문 안에서 확정적으로 초기화되는 변수이기 때문에 굳이 `defaultValue`를 참조해서 초기화 해줄 필요가 없습니다. 반환을 위해서라면 var를 쓰지 않으면 되는 일입니다.~~

Reply:
```cs
private int GetRandomVerticalWide(int min, int max, int startLimit, int endLimit, int defaultValue = 0)
    {
        int wide;
        var borderLimit = GetBorderLimit(BorderSizeY / 2);

        for (var i = 0; i < 3; i++)
        {
            // 임의의 통로 범위 설정
            wide = GetRandomWide(min, max + 1);
            // for문 생략
        }
        return wide;
    }
```
다음과 같이 수정하였을 경우 retuen wide에서 오류 CS0165 할당되지 않은 'wide' 지역 변수를 사용했습니다 가 발생합니다.
개발 당시에도 for문이 들어가는건 확정적인데 어째서인지 다음의 오류가 떠서 defaultValue를 대입해 주었습니다.
이에 대한 해결 방법이 있을까요?

Reply:
생각해보니 그 부분은 답이 없네요. 해당 리뷰는 무시하기 바랍니다.